### PR TITLE
Feat/transcoder event listener

### PR DESCRIPTION
**Pull request by dhruv2295**
Log Exceptions in decoder at 
configure
start
stop & release calls.

information logged at this state is the codec name, media format, and the exception itself.

### CODE REVIEW

File: lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt [hunk 1]
```diff
@@ -283,7 +286,13 @@ class Decoder(
...
         log.i("release(): releasing codec. dequeuedInputs=$dequeuedInputs dequeuedOutputs=$dequeuedOutputs")
-        codec.stop()
-        codec.release()
+        try {
+            codec.stop()
+            codec.release()
+        }
+        catch (e : Exception) {
+            eventListener?.onDecoderReleaseFailure(codec.name, format, e)
+            throw e
```
Comment (sahilbajaj) [inline lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt:295]: why throw the exception here?

File: lib/src/main/java/com/otaliastudios/transcoder/internal/codec/TranscoderEventsListener.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+interface TranscoderEventsListener {
+
+    fun onDecoderConfigureFailure(codecName: String, format: MediaFormat, exception: Exception)
+
+    fun onDecoderStartFailure(codecName: String, format: MediaFormat, exception: Exception)
+
+    fun onDecoderReleaseFailure(codecName: String, format: MediaFormat, exception: Exception)
+
+}
```
Comment (sahilbajaj) [inline lib/src/main/java/com/otaliastudios/transcoder/internal/codec/TranscoderEventsListener.kt:13]: Please check if there's a blank line at the end of the file

**Review by sahilbajaj — CHANGES_REQUESTED** [review]
.

File: lib/src/main/java/com/otaliastudios/transcoder/ThumbnailerOptions.kt [hunk 3]
```diff
@@ -82,6 +85,9 @@ class ThumbnailerOptions(
         fun setListener(listener: ThumbnailerListener) = this.also {
             this.listener = listener
         }
+        fun setEventListener(listener: TranscoderEventsListener) = this.also {
```
Comment (sahilbajaj) [inline lib/src/main/java/com/otaliastudios/transcoder/ThumbnailerOptions.kt:88]: blank line missing above

File: lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt (hunk 1 above)
Comment (dhruv2295) [reply lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt:295]: So that it reaches thumbnail engine and it can use MMR as fallback

**Review by sahilbajaj — APPROVED** [review]

**Pull request merged**